pulsar-client/src/main/java/org/apache/pulsar/client/impl/ProducerImpl.java

```diff
@@ -889,8 +897,15 @@ private void tryRegisterSchema(ClientCnx cnx, MessageImpl msg, SendCallback call
                 Throwable t = FutureUtil.unwrapCompletionException(ex);
                 log.warn("[{}] [{}] GetOrCreateSchema error", topic, producerName, t);
                 if (t instanceof PulsarClientException.IncompatibleSchemaException) {
-                    msg.setSchemaState(MessageImpl.SchemaState.Broken);
-                    callback.sendComplete(t, null);
```

Comment:
Main change 1:  

- The original implementation called `OpSendMsg.callbaclk.sendComplete()` here, it triggers a `recycle` of `OpSendMsg.callback`, but the object `OpSendMsg` is still in `producer. pendingMessages`, which may cause `OpSendMsg.callback` will be concurrently used.